controlplane/internal/controller/imageregistry/image_registry.go

```diff
@@ -0,0 +1,199 @@
+package imageRegistry
```

Comment:
probably this should not be under controller, but at the same level.
also package names should not be camelcase, rather `imageregistry`

Reply:
to clarify, not `controlplane/internal/controller` but `controlplane/internal/imageregistry`

Reply:
Ah gotcha, thanks for clarifying! Changed it in this PR https://github.com/openshift-assisted/cluster-api-agent/pull/24